src/main/webapp/app/faq/faq-update.component.ts

```diff
@@ -0,0 +1,153 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { HttpErrorResponse, HttpResponse } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AlertService } from 'app/core/util/alert.service';
+import { onError } from 'app/shared/util/global.utils';
+import { ArtemisNavigationUtilService } from 'app/utils/navigation.utils';
+import { faBan, faQuestionCircle, faSave } from '@fortawesome/free-solid-svg-icons';
+import { FormulaAction } from 'app/shared/monaco-editor/model/actions/formula.action';
+import { FAQ, FAQState } from 'app/entities/faq.model';
+import { FAQService } from 'app/faq/faq.service';
+import { TranslateService } from '@ngx-translate/core';
+import { FAQCategory } from 'app/entities/faq-category.model';
+import { loadCourseFaqCategories } from 'app/faq/faq.utils';
+import { ArtemisMarkdownEditorModule } from 'app/shared/markdown-editor/markdown-editor.module';
+import { ArtemisCategorySelectorModule } from 'app/shared/category-selector/category-selector.module';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { ArtemisSharedComponentModule } from 'app/shared/components/shared-component.module';
+
+@Component({
+    selector: 'jhi-faq-update',
+    templateUrl: './faq-update.component.html',
+    styleUrls: ['./faq-update.component.scss'],
+    standalone: true,
+    imports: [ArtemisSharedModule, ArtemisSharedComponentModule, ArtemisMarkdownEditorModule, ArtemisCategorySelectorModule],
+})
+export class FAQUpdateComponent implements OnInit {
+    faq: FAQ;
+    isSaving: boolean;
+    existingCategories: FAQCategory[];
+    faqCategories: FAQCategory[];
+
+    domainActionsDescription = [new FormulaAction()];
+
+    // Icons
+    faQuestionCircle = faQuestionCircle;
+    faSave = faSave;
+    faBan = faBan;
+
+    constructor(
+        protected alertService: AlertService,
+        protected faqService: FAQService,
+        protected activatedRoute: ActivatedRoute,
+        private navigationUtilService: ArtemisNavigationUtilService,
+        private router: Router,
+        private translateService: TranslateService,
+    ) {}
+
+    /**
+     * Life cycle hook called by Angular to indicate that Angular is done creating the component
+     */
+    ngOnInit() {
```

Comment:
We don't usually document this 

```suggestion
    ngOnInit() {
```